modules/apps/ccv/types/expected_keepers.go

```diff
@@ -0,0 +1,17 @@
+package types
+
+// StakingKeeper defines the contract expected by parent-chain ccv module.
+// StakingKeeper is responsible for keeping track of latest validator set of all baby chains
+type StakingKeeper interface {
+	GetValidatorSetChanges(chainID string) []abci.ValidatorUpdate
+	// This method is not required by CCV module explicitly but necessary for init protocol
+	GetInitialValidatorSet(chainID string) []sdk.Tx
+}
+
+// CNSKeeper defines the contract expected by parent-chain ccv module
+// CNSKeeper is responsible for verifying that chain creator is authorized to create a chain with given chain-id
+type CNSKeeper interface {
+	AuthorizeChainCreator(chainID, creator string)
```

Comment:
`creator` is a string to support different address formats (i.e bech32 HRP)?

Reply:
Correct

Reply:
I suggest separating the authorization layers from the core ccv functionality. There are a few modules which are increasing the number of paths regarding authentication and authorization, and defining an auth/auth API for ccv now could lead to some re-work later.